distributed/dashboard/components/scheduler.py

```diff
@@ -735,6 +735,101 @@ def name(address):
             update(self.source, result)
 
 
+class WorkerNetworkBandwidth(DashboardComponent):
+    """Worker network bandwidth chart
+
+    Plots horizontal bars with the read_bytes and write_bytes worker state
+    """
+
+    def __init__(self, scheduler, **kwargs):
+        with log_errors():
+            self.scheduler = scheduler
+            self.source = ColumnDataSource(
+                {
+                    "y_read": [],
+                    "y_write": [],
+                    "x_read": [],
+                    "x_write": [],
+                }
+            )
+            self.root = figure(
+                title="Worker Network Bandwidth",
+                tools="",
+                id="bk-worker-net-bandwidth",
+                name="worker_network_bandwidth",
+                **kwargs,
+            )
+
+            # read_bytes
+            self.root.hbar(
+                y="y_read",
+                right="x_read",
+                line_color=None,
+                left=0,
+                height=0.5,
+                fill_color="red",
+                legend_label="read",
+                source=self.source,
+            )
+
+            # write_bytes
+            self.root.hbar(
+                y="y_write",
+                right="x_write",
+                line_color=None,
+                left=0,
+                height=0.5,
+                fill_color="blue",
+                legend_label="write",
+                source=self.source,
+            )
+
+            self.root.axis[0].ticker = BasicTicker(**TICKS_1024)
+            self.root.xaxis[0].formatter = NumeralTickFormatter(format="0.0 b")
+            self.root.xaxis.major_label_orientation = XLABEL_ORIENTATION
+            self.root.xaxis.minor_tick_line_alpha = 0
+            self.root.x_range = Range1d(start=0)
+            self.root.yaxis.visible = False
+            self.root.ygrid.visible = False
+            self.root.toolbar_location = None
+            self.root.yaxis.visible = False
+
+    @without_property_validation
+    def update(self):
+        with log_errors():
+            workers = list(self.scheduler.workers.values())
+
+            y_read = list(np.arange(len(workers)) + 0.75)
+            y_write = list(np.arange(len(workers)) + 0.25)
+
+            x_read = []
+            x_write = []
+            max_limit = 0
+            limit = 104857600  # 100MB
+
+            for ws in workers:
+                x_read.append(ws.metrics["read_bytes"])
+                x_write.append(ws.metrics["write_bytes"])
+
+                max_limit = max(
+                    0.95 * max_limit,
+                    limit,
+                    ws.metrics["read_bytes"],
+                    ws.metrics["write_bytes"],
+                )
+
```

Comment:
```suggestion
            max_limit = max(
                x_read,
                x_write,
                100_000_000,
                self.root.x_range.end,
            )

```

Let's compute this outside of the for loop, and compare against the old version of the x_range.

Reply:
The 95% thing meant to drop down a bit from the last cycle of running this function, not the last worker's memory

Reply:
Running this snippet around 3 times 

 ```python
 import dask.array as da
x = da.random.random((10000, 10000))
y = (x + x.T).compute()

 ```
Previous version: I see that it resets back to ~100MB

![old version](https://user-images.githubusercontent.com/7526622/126671436-469b4cb5-3511-465f-9585-64ee987bcbc3.gif)

 

With the change you suggest, before doing any computation we have the x_range max to be around 100MiB. Then if I run the computation once I see that the x_range limit goes up to 1.3 GB when it finishes it stays in there. If I run it again, the max will go up to 2 GB and stays there (not sure if this is something we desire) 

![matts_version](https://user-images.githubusercontent.com/7526622/126670995-e195fe37-54c3-492c-9ad3-ec5a97d98710.gif)

Reply:
For the record haven't pushed any changes with your suggestion yet, I tested this locally, trying to see what behavior do we want to capture.

Reply:
It looks like I forgot the `0.95 * ` on the self.root.x_range.end bit.

Reply:
I added the `0.95 *`  and that causes that after a computation it'll readjust until the max is 100MB, it'll update few times.  I don't know if this is a problem or not, I don't have a strong opinion on it. 

![update](https://user-images.githubusercontent.com/7526622/126680960-fa927242-e114-4c9f-91a0-34a1c9ca55c3.gif)


I included in the last push labels to the existing bandwidth time-series under /system to be consistent.